pkg/catalog/manager/traverse.go

```diff
@@ -314,7 +347,43 @@ type catalogYml struct {
 func setCatalogErrorState(cmt *CatalogInfo, catalog *v3.Catalog, projectCatalog *v3.ProjectCatalog, clusterCatalog *v3.ClusterCatalog) {
 	v3.CatalogConditionRefreshed.False(catalog)
 	v3.CatalogConditionRefreshed.Message(catalog, fmt.Sprintf("Error syncing catalog %v", catalog.Name))
+	v3.CatalogConditionProcessed.True(catalog)
 	cmt.catalog = catalog
 	cmt.projectCatalog = projectCatalog
 	cmt.clusterCatalog = clusterCatalog
 }
+
+func validVersionURL(versionURL string) bool {
+	if strings.HasPrefix(versionURL, "local://") || strings.HasPrefix(versionURL, "file://") {
+		return false
+	}
+	return true
+}
+
+func getVersionsWithValidURL(urls []string) []string {
+	var validURLs []string
+	for _, url := range urls {
+		if validVersionURL(url) {
+			validURLs = append(validURLs, url)
+		}
```

Comment:
Shouldn't you bubble up an error here ? I see part above that has if 0 valid urls create an error, but do they need to know which error ?

Reply:
https://github.com/rancher/rancher/blob/2e612b488c6ff6cb4bc3313411f34e16b641a0fb/pkg/catalog/manager/crd.go#L95. It gets bubbled here